### ISSUE TITLE
Sorting by and displaying the due date

### ISSUE THREAD

**Issue by Wastus** [basementuniverse/vscode-kanbn#9]
I'm just trying out this extension to organize my life, so there might be a few suggestions coming.

The first one I have, is to allow sorting of the columns by due date. I'm not sure on what basis they are sorted currently, but apparently not by due date. I would find it immensely helpful if the tasks could be sorted by due date.

Another thing in the same direction is, that I don't really care about the update date being shown on the cards. Again the due date would be more helpful to quickly see if I should be working on something rather soon.
I know I can hover over the date to see all date entries, but this is rather slow and takes way to long for a list with many tasks.

But this extension looks really promising.

**Comment by gordon-winkel** [comment]
Hi @Wastus, thanks for the suggestions, and I totally agree that being able to sort columns would be helpful. Currently the columns are sorted according to whatever order the tasks appear in `index.md` (in other words, they're not sorted at all).

The [kanbn CLI app](https://github.com/basementuniverse/kanbn) does allow columns to be sorted (if you've got it installed, try running `kanbn sort --help` for more info). I just haven't implemented a UI for this command in the VSCode extension. It's definitely something that could easily be added though.

As for showing the update date on cards, you make a good point - the 'last updated date' isn't exactly super-important information. In the next release I'll implement a quick-and-dirty method of customising card layout: each card will have elements containing the task's properties (created date, due date, assigned user, tags etc...), and these elements can be shown or hidden using the `board.css` file.

Currently you could use this technique to hide the updated date, like so:

`./kanbn/board.css`:
```
.kanbn-task-date {
  display: none;
}
```

But there isn't a way (yet) to replace it with the due date. If every card contained elements for _all_ of the task's properties (plus default CSS `display` directives for each one, so the default view isn't too crowded), then users could choose exactly what gets displayed for each task.

I'll try to include these features in the next release :+1:

**Comment by basementuniverse** [comment]
I have released version `0.11.0` which adds column sorting and more customisation options for task cards.

There is now a button above each column that allows you to sort the tasks. When you sort a column, you can choose if you want to save the sort settings; if you select 'Yes', the column will automatically sort tasks when a task is moved into/out of the column. By choosing 'No' you can sort the column once without enabling auto-sort.

Also, task cards now contain a lot more information about the task. Some of this information is hidden by default, but you can modify how it looks by adding a custom board stylesheet (`./kanbn/board.css`) - here's how you can hide the created date and show the due date instead:

```
.kanbn-task-data-created {
    display: none;
}

.kanbn-task-data-due {
    display: inline-block;
}
```

There is more information about styling your Kanbn board here:
https://github.com/basementuniverse/vscode-kanbn/blob/main/docs/styles.md

Thanks for your feedback! :smile:
